.editorconfig

```diff
@@ -185,28 +185,53 @@ visual_basic_style_unused_value_assignment_preference = unused_local_variable:wa
 ### Configuration for IDE code style by diagnostic IDs ###
 [*.{cs,vb}]
 
-# IDE0004: Remove unnecessary cast
-dotnet_diagnostic.IDE0004.severity = warning
-
-# IDE0005: Remove unnecessary usings/imports
-dotnet_diagnostic.IDE0005.severity = warning
-
-# IDE0051: Remove unused private members (no reads or writes)
-dotnet_diagnostic.IDE0051.severity = warning
-
-# IDE0052: Remove unread private members (writes but no reads)
-dotnet_diagnostic.IDE0052.severity = warning
-
-# IDE0055: Fix formatting
-dotnet_diagnostic.IDE0055.severity = warning
+dotnet_analyzer_diagnostic.category-Style.severity = warning
```

Comment:
@dotnet/dotnet-analyzers - this PR brings in changes from 2.9.x branch into master which enforces majority of IDE code style rules as build warnings. This is an aggressive route that we are taking to help dogfood our new IDE CodeStyle NuGet package for build time enforcement for consistent code styles. If this turns out too noisy or opinioniated or hindrance to your development experience, please let @sharwell or me know about it, and we can ensure that we tune the code style analyzer, fixer and/or our repo's configuration to meet everyone's preferences. Thank you in helping us dogfood this feature!